contrib/zip/wrapper/wrapper_zip.cc

```diff
@@ -0,0 +1,117 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "contrib/zip/wrapper/wrapper_zip.h"
+
+#include <fcntl.h>
+#include <unistd.h>
+
+#include <iostream>
+
+#include "absl/cleanup/cleanup.h"
+
+constexpr size_t kFileMaxSize = 1024 * 1024 * 1024;  // 1GB
+
+void* zip_source_filefd(zip_t* archive, int fd, const char* mode,
+                        zip_uint64_t start, zip_int64_t len) {
+  FILE* pfile;
+
+  // The file stream is closed when the source is being freed,
+  // usually by zip_close(3).
+  pfile = fdopen(fd, mode);
+  if (pfile == nullptr) {
+    return nullptr;
+  }
+
+  return zip_source_filep(archive, pfile, start, len);
+}
+
+void* zip_source_filefd_create(int fd, const char* mode, zip_uint64_t start,
+                               zip_int64_t len, zip_error_t* ze) {
+  FILE* pfile;
+
+  pfile = fdopen(fd, mode);
+  std::cerr << fd;
+  if (pfile == nullptr) {
+    return nullptr;
+  }
+
+  return zip_source_filep_create(pfile, start, len, ze);
+}
+
+off_t FDGetSize(int fd) {
+  off_t size = lseek(fd, 0, SEEK_END);
+  if (size < 0) {
+    return -1;
+  }
+  if (lseek(fd, 0, SEEK_SET) < 0) {
+    return -1;
+  }
+
+  return size;
+}
+
+void* zip_read_fd_to_source(int fd, zip_error_t* ze) {
+  off_t sizein = FDGetSize(fd);
+  if (sizein > kFileMaxSize) {
+    return nullptr;
+  }
+
+  int8_t* buf = (int8_t*)malloc(sizein);
```

Comment:
Do not use C-style casts. Use a `reinterpret_cast<>`

Reply:
Fixed.